File: Diamond.js
const number = 7;
let r = 1; // row 
let c; // column

// Upper part of the diamond
do {
    c = 1;

    // Printing spaces
    do {
        process.stdout.write(" ");
    }
     while (++c <= number - r + 1);

    c = 1;

    // Printing stars
    do {
        process.stdout.write("*");
    }
    
    while (++c <= r * 2 - 1);

    console.log();
} while (++r <= number);

r = number - 1;

// Lower part of the diamond
do {
    c = 1;
    let n = 1;

    do {
        process.stdout.write(" ");
    } while (++n <= number - r + 1);

    c = 1;

    do {
        process.stdout.write("*");
    } while (++c <= r * 2 - 1);

    console.log();
} while (--r > 0);

File: Diamond.py
n = int(input("Enter the number of rows: "))  

space = n - 1

# Upper part of the diamond
for i in range(1, n + 1):
    for j in range(1, space + 1):
        print(" ", end="") 
    space-=1
    for j in range(1, 2 * i):
        print("*", end="") 
    print("")

space = 1

# Lower part of the diamond
for i in range(1, j): 
    for j in range(i, space+1):
        print(" ", end="")
    space+=1
    for j in range(1, 2 * (n - i)):
        print("*", end="") 
    print("")